### DISCUSSION TITLE
How can I connect to postgres via a unix socket?

### DISCUSSION THREAD

**Discussion by cameron-martin** [ory/kratos#1102] [Q&A] [answered]
Is this possible? If so, what format does the DSN have to be in to do this?

Edit: After a bit of investigation, I've found a [relevant issue](https://github.com/gobuffalo/pop/issues/412).

**Comment by zepatrik** [comment, marked as answer]
I guess you answered the question yourself, we just pass the DSN on to pop.

**Reply by aeneasr** [reply]
@cameron-martin maybe you could post your solution here, if others are looking for this :) Thank you for making Ory better!

**Reply by cameron-martin** [reply]
I didn't actually get any further than determining it was an upstream issue. Because it was the easiest way to fix, I connected via TCP instead. I would still be interested in knowing how to connect via a unix socket though, because this would be preferred.

For further context, I am running kratos in GCP's Cloud Run and unix sockets are preferred because you can mount a cloud sql instance as a unix socket without having to set up a bunch of networking stuff.

I might dig into this in the future, but it's not a priority right now because my app is very much in its infancy.

**Reply by aeneasr** [reply]
Looks like this *should* work: `postgres://?path=/to/socket&database=my-db`

**Reply by cameron-martin** [reply]
Thanks, I'll try it out. Do you know how you would specify a user and password in this format? I guess they can be added using `user` and `password` query params?

**Reply by cameron-martin** [reply]
This doesn't seem to work. I get the following error:

```
2021-03-01 18:36:55.987 GMTtime=2021-03-01T18:36:55Z level=warning msg=Unable to ping database, retrying. audience=application error=map[message:failed to connect to `host=/tmp user=kratos database=kratos`: dial error (dial unix /tmp/.s.PGSQL.5432: connect: no such file or directory)] service_name=kratos service_version=
```

**Reply by zepatrik** [reply]
Probably still the same URL parsing issue in pop then. I assume that if they would pass it on correctly, the driver would handle it.

**Reply by aeneasr** [reply]
Are you sue that the path to the socket is correct? Aren't they usually in `/var/run/postgresql`? It explicitly states `connect: no such file or directory`

**Reply by cameron-martin** [reply]
Cloud run mounts it at `/cloudsql/<instance_connection_name>`. I don't know where it's getting `/tmp/.s.PGSQL.5432` from. Maybe that's what it defaults to?

**Reply by aeneasr** [reply]
What DSN did you use?

**Reply by cameron-martin** [reply]
Unfortunately I removed the DSN from the secrets store immediately after I tested it. If I get around to trying it again I'll post it here, but it's not a priority for me right now.
